src/libsync/owncloudpropagator.cpp

```diff
@@ -386,6 +456,17 @@ void OwncloudPropagator::start(const SyncFileItemVector& items)
                 currentDirJob->append(dir);
             }
             directories.push(qMakePair(item->destination() + "/" , dir));
+        } else if (enableBundledRequests
+                  && (item->_instruction == CSYNC_INSTRUCTION_NEW)
+                  && (item->_direction == SyncFileItem::Up)
+                  && (item->_size < chunkSize)) {
+            //this will create list of bundle files to sync for that bundlejob
+            if (directories.top().second->_bundleJob.isNull()) {
+                PropagateBundle* bundleJob = new PropagateBundle(this);
+                directories.top().second->_bundleJob.reset(bundleJob);
+            }
+            PropagateBundle* bundleJob = dynamic_cast<PropagateBundle*>(directories.top().second->_bundleJob.data());
```

Comment:
Arent we using dynamic_cast for Directories? https://github.com/owncloud/client/blob/master/src/libsync/owncloudpropagator.cpp#L317 Line 317